### DISCUSSION TITLE
Using PyLHE as a filter for rewriting

### DISCUSSION THREAD

**Discussion by agbuckley** [scikit-hep/pylhe#238] [Ideas]
Hi all, I appreciate the focus on _reading_ LHE files, but it would be super-useful to also be able to write them -- in particular for fixing or filtering slightly borked files, which isn't that uncommon. (And this reminds me that being able to convert to and from the LHEH5 HPC format would be very useful as a further extension.)  LHE is a pretty simple format, but does anyone have any simple output code already written, based upon PyLHE read-in? Thanks!

**Comment by matthewfeickert** [comment]
Good point @agbuckley. I don't think that anyone is against having this functionality, just that @lukasheinrich originally wrote it to read and no one has had the time to add new functionality for a bit.

> LHE is a pretty simple format, but does anyone have any simple output code already written, based upon PyLHE read-in?

I don't think so(?), but @APN-Pucky has started PR https://github.com/scikit-hep/pylhe/pull/233 (I think that's waiting on me to review :scream:) that adds _event_ writing.

**Reply by APN-Pucky** [reply]
Yes, I updated it for newer flake8 checks now. Last time codecov complained about not having tests that probe a except block, iirc. I was not sure if I should also add a wrong test file that fails to increase the coverage.

**Reply by agbuckley** [reply]
Ah, excellent! That looks very usable: either by holding all the modified/filtered events in memory and calling `write_lhe_string/file()` or in a more "streaming" way by calling the manual `.tolhe()` methods for the init object and then each event in turn.

Any ETA on when this might be publicly available? I don't mind checking out dev branches personally, but the immediate "clients" of this functionality (an ATLAS analysis team asking for workarounds on Pythia aborts of sometimes borked LHE inputs) would need it to be `pip`able. Thanks!

**Reply by matthewfeickert** [reply]
> Any ETA on when this might be publicly available?

@agbuckley We could probably get a minor release out (given that PR https://github.com/scikit-hep/pylhe/pull/233 is a new feature) by the end of the week or start of next, but given that I'm entering crunch for writing, workshop organization, and travel I will really _need_ people to be "squeaky wheels" to help that. :)

**Reply by matthewfeickert** [reply]
@agbuckley, [`v0.9.0` is out now on PyPI](https://pypi.org/project/pylhe/0.9.0/) (and will be out on conda-forge soon too).

c.f. Discussion https://github.com/scikit-hep/pylhe/discussions/249

**Reply by eduardo-rodrigues** [reply]
(Catching up on backlog.)

This can now be closed modulo my [comment on the MR](https://github.com/scikit-hep/pylhe/pull/233#issuecomment-2309825784), I believe, @APN-Pucky.

**Reply by matthewfeickert** [reply]
I've migrated that to Issue https://github.com/scikit-hep/pylhe/issues/251 for tracking.
